src/main/java/life/qbic/registration/openbis/OpenBisDropboxETL.java

```diff
@@ -0,0 +1,87 @@
+package life.qbic.registration.openbis;
+
+import static java.util.Objects.requireNonNull;
+
+import ch.systemsx.cisd.common.exceptions.NotImplementedException;
+import ch.systemsx.cisd.etlserver.registrator.DataSetRegistrationContext;
+import ch.systemsx.cisd.etlserver.registrator.api.v2.AbstractJavaDataSetRegistrationDropboxV2;
+import ch.systemsx.cisd.etlserver.registrator.api.v2.IDataSet;
+import ch.systemsx.cisd.etlserver.registrator.api.v2.IDataSetRegistrationTransactionV2;
+import ch.systemsx.cisd.openbis.dss.generic.shared.api.internal.v2.ISampleImmutable;
+import java.io.File;
+import life.qbic.registration.openbis.exceptions.fail.MeasurementHasDataException;
+import life.qbic.registration.openbis.exceptions.fail.UnknownSampleTypeException;
+import life.qbic.registration.openbis.types.QDatasetType;
+import life.qbic.registration.openbis.types.QPropertyType;
+import life.qbic.registration.openbis.types.QSampleType;
+
+/**
+ * The Dropbox ETL process.
+ * <p>
+ * This process is responsible for
+ *  <ol>
+ *    <li>fetching the measurement sample from openbis
+ *    <li>creating a data set linked to the sample
+ *    <li>moving files into the dataset
+ *  </ol>
+ *  Some constraints are taken care of during this process. These constraints being:
+ *  <ul>
+ *    <li>the measurement sample already exists in openbis
+ *    <li>only one measurement sample exists with the provided identifier
+ *    <li>the measurement sample has no data set linked as one measurement can only have one dataset.
+ *  </ul>
+ */
+public class OpenBisDropboxETL extends AbstractJavaDataSetRegistrationDropboxV2 {
+
+  private static final String PROVENANCE_FILE_NAME = "provenance.json";
```

Comment:
maybe collect it with the parser in one class, so the provenance concept has higher cohesion

Reply:
This was my first idea. I though about it and I decided to move it to a parameter of the parser as the filename does not have any impact on the parsing. This way the DropboxETL decides on the way the data is expected to arrive. So if we want to change the location or name of the file, this is done in the overall process and the parsing logic is not really impacted. I think that locating the filename in this class makes sense as the overall flow (look for a file `provenance.json`, search measurement, does it fit expected properties?, create dataaset) is easier read in the process method.

Reply:
Fair enough :)